admin/server/organizations.go

```diff
@@ -5,27 +5,47 @@ import (
 	"errors"
 
 	"github.com/rilldata/rill/admin/database"
+	"github.com/rilldata/rill/admin/server/auth"
 	adminv1 "github.com/rilldata/rill/proto/gen/rill/admin/v1"
 	"google.golang.org/grpc/codes"
 	"google.golang.org/grpc/status"
-	timestamppb "google.golang.org/protobuf/types/known/timestamppb"
+	"google.golang.org/protobuf/types/known/timestamppb"
 )
 
 func (s *Server) ListOrganizations(ctx context.Context, req *adminv1.ListOrganizationsRequest) (*adminv1.ListOrganizationsResponse, error) {
+	// Check the request is made by an authenticated user
+	claims := auth.GetClaims(ctx)
+	if claims.OwnerType() != auth.OwnerTypeUser {
+		return nil, status.Error(codes.Unauthenticated, "not authenticated as a user")
+	}
+
 	orgs, err := s.admin.DB.FindOrganizations(ctx)
 	if err != nil {
 		return nil, status.Error(codes.InvalidArgument, err.Error())
 	}
+	// filter out orgs that the user doesn't have access to
+	filteredOrgs := make([]*database.Organization, 0, len(orgs))
+	for _, org := range orgs {
+		if claims.CanOrganization(ctx, org.ID, auth.ReadOrg) {
+			filteredOrgs = append(filteredOrgs, org)
+		}
+	}
```

Comment:
This will pull all organizations and filter client-side. It should use a new DB query that filters in SQL based on memberships (to leverage indexes and not load all orgs into memory.)

Reply:
done